src/main/java/org/opentripplanner/netex/loader/parser/NetexDocumentParser.java

```diff
@@ -101,22 +100,23 @@ private void parseCompositeFrame(CompositeFrame frame) {
     }
   }
 
-  private void parseFrameDefaultsLikeTimeZone(VersionFrameDefaultsStructure frameDefaults) {
-    String timeZone = "GMT";
+  private <T> void parse(T node, NetexParser<T> parser) {
+    parser.parse(node);
+    parser.setResultOnIndex(netexIndex);
 
-    if (
-      frameDefaults != null &&
-      frameDefaults.getDefaultLocale() != null &&
-      frameDefaults.getDefaultLocale().getTimeZone() != null
-    ) {
-      timeZone = frameDefaults.getDefaultLocale().getTimeZone();
+    if (node instanceof VersionFrame_VersionStructure frame) {
+      netexIndex.timeZone.set(resolveTimeZone(frame.getFrameDefaults()));
     }
-
-    netexIndex.timeZone.set(timeZone);
   }
 
-  private <T> void parse(T node, NetexParser<T> parser) {
-    parser.parse(node);
-    parser.setResultOnIndex(netexIndex);
+  private String resolveTimeZone(VersionFrameDefaultsStructure frameDefaults) {
+    return Optional
+      .ofNullable(frameDefaults)
+      .map(VersionFrameDefaultsStructure::getDefaultLocale)
+      .map(LocaleStructure::getTimeZone)
+      // Fallback to previously set time zone in hierarchy
+      .or(() -> Optional.ofNullable(netexIndex.timeZone.get()))
+      // Fallback to GMT if no time zone exists in hierarchy
+      .orElse("GMT");
```

Comment:
I disagree respectfully.

Personally, I find the forest of null checks unreadably ugly and leading to a lot of duplicated variable names.

Reply:
Could you explain why you feel that Optional was not made for this?

Reply:
IMO, it was made _exactly_ for this.

Reply:
It was originally meant for output types for specific functions, as described in this blog post https://nipafx.dev/design-java-optional/

I personally agree with Leonard, that usage of `Optional` looks much more readable in cases, where we need to have long call chains, where multiple places in the chain can produce null. Hopefully this will eventually get resolved when JEPs 401 and 402 from [project Valhalla](https://openjdk.org/projects/valhalla/) land in the future.

Reply:
We discussed this before (spring I believe). @leonardehrenfried can you point to an article which say that this is more readable, after all, you argue to break a basic principle? I have posted links to similar posts as @hannesj before.

I recently went to an open-space discussion about OO vs Functional programming and asked about this. I was a bit surprised about the reaction. Two of the programmers there, sad this was NOT functional programming, and meant this was the "when you got a new hammer, everything looks like a nail".  Further, they went into a discussion about type-safety, which I did not understand fully - except that the "recommended way" for using Optional (and other functional interfaces in Java) was to return you own types, not the library - to provide type-safety and get a "natural business language". For example the "Function<T,S>" do a transform when you call `apply`, but `apply` is almost the wrong verb in every case. So you should make you own type. Streams/call chains are very good for filtering and transformations, but I find it hard to read branching.

Reply:
Another argument here is tool support, static analyses will catch mistakes in if...else blocks.